nixos/tests/quotas.nix

```diff
@@ -0,0 +1,72 @@
+import ./make-test-python.nix ({pkgs, ... }:
+{ 
+  name = "quotas";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ zupo ];
+  };
+  nodes.machine = {
+    virtualisation = {
+      emptyDiskImages = [ 1 ];
+      fileSystems = {
+        "/mnt" = {
+          device = "/dev/vdb";
+          fsType = "ext4";
+          autoFormat = true;
+          options = [
+            "defaults"
+            # Not sure why this is necessary..
+            "usrquota"
+            "grpquota"
+          ];
+        };
+      };
+    };
+
+    quotas = {
+      enable = true;
+      fileSystems = {
+        "/mnt" = {
+          enable = true;
+          userquotas = {
+            root = {
+              enable = true;
+              block-softlimit = 100;
+              block-hardlimit = 200;
+              inode-softlimit = 300;
+              inode-hardlimit = 400;
+            };
+          };
+        };
+      };
+    };
+  };
+
+  # Disable linting for simpler debugging of the testScript
+  skipLint = true;
+
+  testScript = ''
+    start_all()
+
+    out = machine.succeed("cat /etc/fstab")
+    print(out)
+    assert "usrquota" in out
+    assert "grpquota" in out
+
+
+    exit, out = machine.systemctl("cat set-disk-quotas.service")
+    print(out)
+    exit, out = machine.systemctl("start set-disk-quotas.service")
+    print(out)
+    if exit != 0:
+      out = machine.succeed("journalctl -u set-disk-quotas.service")
+      print(out)
+      exit(1)
```

Comment:
not sure what the nixos custom is, but the python docs say this about the exit builtin (and similar builtins):

> They are useful for the interactive interpreter shell and should not be used in programs.

https://docs.python.org/3/library/constants.html

Reply:
Seems like only `nixos/tests/matrix-appservice-irc.nix` uses the exit builtin, and only a couple tests use sys.exit. I think mose use `assert False, "message"` or similar.

Reply:
@CameronNemo Good feedback, I simplified the test.